fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -1749,26 +1765,270 @@ public Status resetIdsForRestore(GlobalStateMgr globalStateMgr, Database db, int
+            newSimpleDefineSql = newSimpleDefineSql.replaceAll(oldDbTableName, newDbTableName);
+        }
+
+        try {
+            analyzeMvDefineQuery(connectContext, newViewDefineDql);
+
+            // only renew defined view sql when analyze the new query success.
+            LOG.info("Renew materialized view' defined sql from {} to {}", name, oldViewDefineSql, newViewDefineDql);
+            LOG.info("Renew materialized view' simple defined sql from {} to {}", name, oldSimpleDefineSql, newSimpleDefineSql);
+            this.viewDefineSql = newViewDefineDql;
+            this.simpleDefineSql = newSimpleDefineSql;
+
+            return true;
+        } catch (Exception e) {
+            String errorMsg = String.format("Can not active materialized view [%s]" +
+                    " because analyze materialized view define sql failed: \n\n%s" +
+                    "\n\nCause an error: %s", name, newViewDefineDql, e.getMessage());
+            LOG.warn(errorMsg);
+        }
+        return false;
+    }
+
+    private boolean isCurrentRemoteMvId(MvId mvId, Map<MvId, MvBackupInfo> mvIdTableNameMap) {
+        if (!mvIdTableNameMap.containsKey(mvId)) {
+            return false;
+        }
+        MvBackupInfo mvBackupInfo = mvIdTableNameMap.get(mvId);
+        if (mvBackupInfo == null) {
+            return false;
+        }
+        MvId localMvId = mvBackupInfo.getLocalMvId();
+        if (localMvId == null) {
+            return false;
+        }
+        // find the local mvId is same the current materialized view
+        return localMvId.getId() == this.id && localMvId.getDbId() == dbId;
+    }
+
+    private Optional<MvId> restoreBaseTable(Database db, Table baseTable, MvRestoreContext mvRestoreContext) {
+        Map<MvId, MvBackupInfo> mvIdTableNameMap = mvRestoreContext.getMvIdToTableNameMap();
+        Set<MvId> mvIds = baseTable.getRelatedMaterializedViews();
+        Optional<MvId> oldMvId = mvIds.stream()
+                .filter(mvId -> isCurrentRemoteMvId(mvId, mvIdTableNameMap))
+                .findFirst();
+        if (!oldMvId.isPresent()) {
+            LOG.warn(String.format("Cannot find base table info used by the table %s.%s in the materialized view %s, " +
+                            "mvIds:%s", db.getFullName(), baseTable.getName(), name,
+                    mvIds.stream().map(x -> x.toString()).collect(Collectors.joining(","))));
+        } else {
+            LOG.info("Remove remote base table info {} used by the table {}.{} in the materialized view {}",
+                    oldMvId.get(), db.getFullName(), baseTable.getName(), name);
+            mvIds.remove(oldMvId.get());
+        }
+
+        // update base table's related mv's info
+        mvIds.add(new MvId(db.getId(), this.getId()));
+        return oldMvId;
+    }
+
+    private void restoreBaseTableVersionMap(Map<Long, Map<String, BasePartitionInfo>> baseTableVisibleVersionMap,
```

Comment:
could you move it to something like `MvRestore****` ?

Reply:
fixed.